### ISSUE TITLE
DD4D doesn't work in Firefox 80.0.1 on Windows

### ISSUE THREAD

**Issue by jamesgecko** [nick-vanpraet/dndbeyonddigitaldicefordiscord#3]
I've been unable to get the Firefox version of this extension working. It displays the blue bar saying a connection is currently active, selects the dice result and `console.log`s the selected `HTMLCollections`. The selected elements appear to be correct. I don't see an error in dev tools, but the network tab shows that the Discord webhook isn't being hit.

I've disabled all my other extensions and Firefox Enhanced Tracking Protection, but it didn't make a difference. The extension _does_ work in Chrome with the exact same configuration.

Any idea what's going wrong?

**Comment by nick-vanpraet** [comment]
Hmm, I can't reproduce this locally. I can send rolls to a channel using the direct discord webhook or via the bot url. 

Although it also doesn't show the network activity, seems Firefox hides it for content scripts. Very annoying.

What version of Firefox do you have, and what version of the addon are you using (release or developer version)?

**Comment by nick-vanpraet** [comment]
Christ, my eyes, version's in the title

**Comment by nick-vanpraet** [comment]
Ok, updated FF to 80.0.1, but still can't reproduce it. If the POST fails, I believe it should show up as an exception in console at least, unless FF hides those for addons' content scripts as well.

**Comment by swichers** [comment]
This might be related to cooking permissions. Posting to the webhook may be attempting to set a cookie from discordapp.com which would be considered a 3rd party/external resource. I had to specifically allow this action to bypass in order to get the extension to work properly.

This was not FF, but just throwing it out there as a possibility because FF has tougher restrictions out of the box.

**Comment by nick-vanpraet** [comment]
Blocking cookies didn't seem to reproduce it, although I may have bungled it.

@jamesgecko Could you enable Browser Toolbox in FF and take a look in that one's Network tab. That one does capture extension network traffic.

https://developer.mozilla.org/en-US/docs/Tools/Browser_Toolbox

**Comment by jamesgecko** [comment]
Automatically updated to Firefox 81.0. No change in behavior. I don't see a request to Discord in the network tab in Browser Toolbox after rolling.

**Comment by d0mmi** [comment]
i tested it in Firefox 84.0.2 and it is also not working for me

**Comment by nick-vanpraet** [comment]
Ok, I can reproduce it now. I spent all evening debugging, but in the end it's just because in the manifest.json permissions "discordapp.com" is listed but discord has moved on and now uses "discord.com" for their webhooks too. Apparently trying to POST to a domain not mentioned in permissions just makes the entire call disappear, no errors, no warnings, just poof.

In any case, I updated manifest.json and released a new FF version. Should be up on the addon store now. Let me know if it fixes it, it fixed it for me.
